### PR TITLE
Uplift Transformers to 5.5.1

### PR DESCRIPTION
**Pull request by ssaliceTT**
### Problem description

Transformers 5.5.1 introduced several breaking changes to the attention and cache APIs that affected model tests, the LLaMA example, and the perf benchmark infra. This PR brings `tt-xla` and related submodules up to date with Transformers 5.5.1. This version of transformers is also required for the next major uplift of vLLM

---

### Breaking changes in Transformers 5.5 and how it's handled

**`cache_position` deprecated → `position_ids`**

`cache_position` is no longer passed to attention modules in 5.5; `position_ids` replaces it. Updated `examples/pytorch/llama.py` and `tests/torch/models/llama3/test_llama_step_n300.py` to pass a 2D `position_ids` tensor instead and remove `cache_position` from all input dicts and device transfers.
_Note:  custom models not registered in transformers such as kimi_k2/kimi_k2_5 and DeepSeek V3.2 still rely on `cache_position` for their cache write paths so it is still left unchanged in `LLMSamplingWrapper`. Standard transformers models accept it via `**kwargs` and ignore it and so are unaffected.

**`StaticCache` lazy initialization now fires during tracing**

In 5.5, `StaticCache` defers internal buffer allocation until the first forward pass. This caused `paged_update_cache` legalization failures. Fixed by calling `static_cache.early_initialization(...)` explicitly across all attention decode tests.

**`StaticCache.cumulative_length` accumulates in-place across runs**

`torch_device_runner.py`'s `to_device` was mutating the source object in-place, causing `cumulative_length` to accumulate across repeated runs of the same workload. Fixed with `copy.copy` on objects with `__dict__` and `.clone()` on tensors that return `self` from `.to()`.

**`return_dict=False` now propagates into sub-models (5.5)**

Setting `return_dict=False` on the top-level OWL-ViT model in 5.5 propagates into `vision_model` and other sub-models, breaking `.vision_model_output` attribute access. Removed `return_dict=False` from the OWL-ViT loader and updated `post_process` to use attribute access (`outputs.logits`, `outputs.pred_boxes`) instead of tuple indexing.

**`GptOssAttention.sinks` uninitialized → NaN in bfloat16**

`sinks` is declared as `nn.Parameter(torch.empty(...))`. After `.to(torch.bfloat16)`, uninitialized float32 bits can produce bfloat16 NaN, corrupting softmax output across all heads. Added `attention.sinks.data.zero_()` in both prefill and decode tests, and added `attention.sinks` to the TP shard spec (`"model"` dim) to match the head sharding. (need to test further)

**GLM4-MoE and Kimi-K2 test fixes**

Minor API changes in 5.5 broke the GLM4-MoE loader and Kimi-K2 prefill test (`cache_positions` arg removed, `lazy_initialization` required before tracing). Both updated accordingly.

**DeepSeek V3.2 Exp tokenizer: `AutoTokenizer` triggers 5.5 `rope_scaling` bug**

`AutoTokenizer.from_pretrained` loads the model config to detect the tokenizer class. For `deepseek_v32` (unregistered in transformers 5.5), this triggers `convert_rope_params_to_dict` before `max_position_embeddings` is set on the config, causing a `KeyError`. Switched to `PreTrainedTokenizerFast.from_pretrained`, which loads `tokenizer.json` directly without touching the model config.

---

### What's changed

- `venv/requirements-dev.txt`: bump `transformers==5.2.0` → `5.5.1`
- `examples/pytorch/llama.py`: `cache_position` → `position_ids`; move `cumulative_length` to device
- `tests/infra/runners/torch_device_runner.py`: `copy.copy` + tensor clone to prevent StaticCache state corruption across runs
- `tests/torch/graphs/test_attention.py`: `early_initialization` for all StaticCache decode tests; `sinks.zero_()` + shard spec fix for GptOss prefill/decode
- `tests/torch/models/llama3/test_llama_step_n300.py`: remove `cache_position`, use `position_ids`
- `tests/torch/models/glm4_moe/test_glm4_moe.py`: 5.5 API compat
- `tests/torch/models/kimi_k2/test_kimi_k2.py`: `cache_positions` removal + `lazy_initialization`
- `tests/torch/models/deepseek_v3_2_exp/test_deepseek_v3_2_exp.py`: `AutoTokenizer` → `PreTrainedTokenizerFast` to avoid 5.5 `rope_scaling` bug
- `tests/benchmark/benchmarks/llm_benchmark.py`, `decode_utils.py`: perf benchmark updated for 5.5 StaticCache API; `cache_position` still kept in `LLMSamplingWrapper` inner model call for compatibility with custom models (kimi_k2/k2_5, DeepSeek V3.2)
- `.github/workflows/perf-bench-matrix.json`: transformer version and benchmark config updates (all models including kimi_k2_5)
- `third_party/tt_forge_models` (submodule): OWL-ViT loader fix for `return_dict` propagation; DeepSeek V3.2 Exp loader fixes for tokenizer loading and Indexer k_cache safety fallback (merged via [tt-forge-models#700](https://github.com/tenstorrent/tt-forge-models/pull/700))

### Checklist
- [x] Merge tt-forge-models changes
- [x] Merge fixes in tt-mlir
- [x] Clean up perf infra changes to remove all `cache_position` references (unless necessary)
- [x] Ensure nightly tests are all on par
- [x] Ensure no major perf regressions

### CODE REVIEW

**Comment by AleksKnezevic** [comment]
@ssaliceTT can you please link to nightly and perf tests.

**Comment by ssaliceTT** [comment]
> @ssaliceTT can you please link to nightly and perf tests. 

Will do soon. Have a few more things to fix up that I'd missed.

**Review by vmilosevic — APPROVED** [review]

**Review by vvukomanTT — APPROVED** [review]

File: third_party/CMakeLists.txt [hunk 1]
```diff
@@ -5,7 +5,7 @@
 option(USE_CUSTOM_TT_MLIR_VERSION "Flag to use TT_MLIR_VERSION set by the user" OFF)
 
 if (NOT DEFINED TT_MLIR_VERSION OR NOT USE_CUSTOM_TT_MLIR_VERSION)
-    set(TT_MLIR_VERSION "412daacc440f10bb98ccc685c311b01f1fadab70")
+    set(TT_MLIR_VERSION "ssalice/fill-cache")
```
Comment (sgligorijevicTT) [inline third_party/CMakeLists.txt]: flagging this so we don't forget

**Review by nsumrakTT — APPROVED** [review]

**Comment by ssaliceTT** [comment]
### Updates 
#### Nightly tests and status
Basic test nightly: https://github.com/tenstorrent/tt-xla/actions/runs/25193730850
model test passing: https://github.com/tenstorrent/tt-xla/actions/runs/25193747475
model test full (older - 3 days ago): https://github.com/tenstorrent/tt-xla/actions/runs/25065364136

Currently everything is on par (with Nightly 540 4-29) except one test that failed in basic test nightly (gpt oss prefill). Fix for it tested separately, looks good. The owl-vit test had been fixed, didn't add the right tt-forge-models commit in the tests above.

#### Perf tests and status
n150 perf: https://github.com/tenstorrent/tt-xla/actions/runs/25194573832
n300-llmbox perf: https://github.com/tenstorrent/tt-xla/actions/runs/25194590982/attempts/1

Mainly looking at n300-llmbox regressions. With fixes in my tt-mlir branch, regression has been brought down significantly from ~20 to ~30% regression on most models to three models being regressed:

- llama_3_1_70b_tp:                               -12.8%  (4.8 → 4.1842 SPS)
- qwen_3_32b_tp:                                   -7.6%  (4.84 → 4.4738 SPS)
- falcon3_10b_tp:                                  -7.2%  (25.75 → 23.8944 SPS)

Currently investigating these as llama 3.1 70b is a priority.

_(Note: all tests were run on a draft branch)_

File: .github/workflows/perf-bench-matrix.json [hunk 2]
```diff
@@ -24,52 +24,52 @@
       },
       {
         "name": "mnist",
-        "pyreq": "datasets loguru pytest requests torch==2.9.0 tqdm transformers==5.2.0",
+        "pyreq": "datasets loguru pytest requests torch==2.9.0 tqdm transformers==5.5.0",
```
Comment (mmanzoorTT) [inline .github/workflows/perf-bench-matrix.json]: It should be 5.5.1 for all the tests. 

```suggestion
        "pyreq": "datasets loguru pytest requests torch==2.9.0 tqdm transformers==5.5.1",
```

**Comment by kmabeeTT** [comment]
@ssaliceTT  - Any updates on rebase and rerun since tt-mlir uplift landed earlier? Let's try and land this cleanly.

**Comment by ssaliceTT** [comment]
> @ssaliceTT - Any updates on rebase and rerun since tt-mlir uplift landed earlier? Let's try and land this cleanly.

Running tests now, will update with results. Expecting parity with nightly.

**Review by AleksKnezevic — APPROVED** [review]

**Review by odjuricicTT — APPROVED** [review]

**Comment by ssaliceTT** [comment]
### CI Updates 
#### Nightly tests and status
basic-test-nightly: https://github.com/tenstorrent/tt-xla/actions/runs/26499178573
model-test-passing: [https://github.com/tenstorrent/tt-xla/actions/runs/26424406609](https://github.com/tenstorrent/tt-xla/actions/runs/26499203606)
model-test-full: https://github.com/tenstorrent/tt-xla/actions/runs/26424428138

Currently tests have parity with Nightly [570](https://github.com/tenstorrent/tt-xla/actions/runs/26425442160) from 5-25. 

#### Perf tests and status
n150: [https://github.com/tenstorrent/tt-xla/actions/runs/26456457942](https://github.com/tenstorrent/tt-xla/actions/runs/26499226252)
n300-llmbox: https://github.com/tenstorrent/tt-xla/actions/runs/26456460499
galaxy-wh-6u: https://github.com/tenstorrent/tt-xla/actions/runs/26485617342

Mainly looked at n300-llmbox perf regressions as previously stated and there are two regressions leftover as follows:
  - falcon3_10b_tp:                                  -7.6%  (23.39 → 21.6155 SPS)
  - falcon3_7b_tp:                                   -5.7%  (31.15 → 29.3622 SPS)

As per @odjuricicTT and my discussions, his team will take care of them. I'll be moving forward with his approval.

_(Note some tests were run with a draft branch instead of the PR branch)_
